Failed to migrate Scripts/ folder: Access is denied.

## Description

> [!NOTE]
> Could only reproduce this issue on Windows so far, but I expect this might happen on Linux as well.

Opening an old Overload project (using the Scripts/ folder, just as Showroom) fail and shows the following error:
```
[ERROR] 2026-04-16_10-14-14 Failed to migrate Scripts/ folder: Access is denied.
```

## To Reproduce
Steps to reproduce the behavior:
1. Download Showroom latest
2. Open Showroom
3. Accept Scripts/ folder migration
4. See error

> [!NOTE]
> If no error, try creating an empty `Scripts/` folder in the project folder **after** a successful migration, and > trying again steps 2 to 4.

## Expected behavior
The migration should succeed.

## Screenshots

<img width="466" height="295" alt="Image" src="https://github.com/user-attachments/assets/0c45a560-0c33-470b-93a2-42fc4b03f34e" />

<img width="831" height="335" alt="Image" src="https://github.com/user-attachments/assets/639f9f2f-a285-483e-bd63-478b33f02a83" />

--- a/Sources/OvEditor/src/OvEditor/Core/EditorActions.cpp
+++ b/Sources/OvEditor/src/OvEditor/Core/EditorActions.cpp
@@ -1166,6 +1166,18 @@ void OvEditor::Core::EditorActions::MigrateScripts()
 	}
 
 	const auto targetPath = m_context.projectAssetsPath / "Scripts";
+	std::vector<std::filesystem::path> migratedScriptNames;
+
+	for (const auto& entry : std::filesystem::recursive_directory_iterator(legacyScriptsPath))
+	{
+		if (!entry.is_directory())
+		{
+			if (OvTools::Utils::PathParser::GetFileType(entry.path().string()) == OvTools::Utils::PathParser::EFileType::SCRIPT)
+			{
+				migratedScriptNames.push_back(entry.path().filename());
+			}
+		}
+	}
 
 	std::error_code err;
 
@@ -1224,18 +1236,12 @@ void OvEditor::Core::EditorActions::MigrateScripts()
 	OVLOG_INFO("Scripts/ folder migrated to Assets/Scripts/");
 
 	// Update all scene files: replace old behaviour type (just the stem) with the new relative path
-	for (const auto& entry : std::filesystem::recursive_directory_iterator(targetPath))
+	for (const auto& scriptName : migratedScriptNames)
 	{
-		if (!entry.is_directory())
-		{
-			if (OvTools::Utils::PathParser::GetFileType(entry.path().string()) == OvTools::Utils::PathParser::EFileType::SCRIPT)
-			{
-				const auto stem = entry.path().stem().string();
-				const auto newRelPath = (std::filesystem::path("Scripts") / entry.path().filename()).generic_string();
+		const auto stem = scriptName.stem().string();
+		const auto newRelPath = (std::filesystem::path("Scripts") / scriptName).generic_string();
 
-				PropagateFileRenameThroughSavedFilesOfType(stem, newRelPath, OvTools::Utils::PathParser::EFileType::SCENE);
-			}
-		}
+		PropagateFileRenameThroughSavedFilesOfType(stem, newRelPath, OvTools::Utils::PathParser::EFileType::SCENE);
 	}
 
 	OVLOG_INFO("Scene files updated with new script paths");

--- a/Sources/OvEditor/src/OvEditor/Core/EditorActions.cpp
+++ b/Sources/OvEditor/src/OvEditor/Core/EditorActions.cpp
@@ -1168,7 +1168,52 @@ void OvEditor::Core::EditorActions::MigrateScripts()
 	const auto targetPath = m_context.projectAssetsPath / "Scripts";
 
 	std::error_code err;
-	std::filesystem::rename(legacyScriptsPath, targetPath, err);
+
+	if (!std::filesystem::exists(targetPath))
+	{
+		std::filesystem::rename(legacyScriptsPath, targetPath, err);
+	}
+	else if (std::filesystem::is_directory(targetPath))
+	{
+		std::filesystem::create_directories(targetPath, err);
+
+		if (!err)
+		{
+			for (const auto& entry : std::filesystem::recursive_directory_iterator(legacyScriptsPath))
+			{
+				const auto destination = targetPath / entry.path().lexically_relative(legacyScriptsPath);
+
+				if (entry.is_directory())
+				{
+					std::filesystem::create_directories(destination, err);
+				}
+				else
+				{
+					std::filesystem::create_directories(destination.parent_path(), err);
+
+					if (!err)
+					{
+						std::filesystem::copy_file(entry.path(), destination, std::filesystem::copy_options::overwrite_existing, err);
+					}
+				}
+
+				if (err)
+				{
+					break;
+				}
+			}
+		}
+
+		if (!err)
+		{
+			std::filesystem::remove_all(legacyScriptsPath, err);
+		}
+	}
+	else
+	{
+		OVLOG_ERROR("Failed to migrate Scripts/ folder: Assets/Scripts exists but is not a directory.");
+		return;
+	}
 
 	if (err)
 	{

